### ISSUE TITLE
Lamda Value for Training EWC model

### ISSUE THREAD

**Issue by ahtsham58** [jasonwu0731/trade-dst#22]
hi @andreamad8 @jasonwu0731,

I am trying to train EWC model. 
`python3 EWC_train.py -bsz=8 -dr=0.2 -lr=0.001 -path=${save_path_except_domain} -exceptd=${except_domain} -fisher_sample=10000 -l_ewc=${lambda}`

What should be the value of `-l_ewc=${lambda}`

In the paper it is mentioned "For EWC, we set different
values of lamda for all the domains, and the optimal
value is selected using the validation set" that doesnt make sense to me, may be I am missing something.

Can you please give me the lamda value that you used..?

**Comment by andreamad8** [comment]
Hi @ahtsham58,

we tried several values: 1000, 100, 10, 1, 0.1, 0.01, 0.001, 0.0001. I remember each of the domain had a different value. I don't recall the exact value, but good performance can be achieved using 0.001 and 0.01. 

I hope this helps. 

Andrea

**Comment by ahtsham58** [comment]
Hi @andreamad8 @jasonwu0731 ,

Really thankful to you guys for your continuous support as I am done with everything and reproduced the system.

You would have thought about why I am doing all this stuff? So here is my answer
Actually, I want to see if this system is really doing what it claims for examples, state tracking in mentioned- domains and unseen domains and how effectively it can recognize the states.

Lastly, Can you please give me an idea, How can I prepare and use a small test dataset with my own customize queries using these models?
Secondly, is it possible to see what type of [slots, values] it extracted from the conversation after the testing ?

**Comment by ahtsham58** [comment]
@andreamad8 @andreamad8 
Could you please answer this.. 
"is it possible to see what type of [slots, values] it extracted from the conversation after the testing?"

**Comment by jasonwu0731** [comment]
hi,

the all_prediction dictionary should have the information you are looking for

https://github.com/jasonwu0731/trade-dst/blob/master/models/TRADE.py#L197
